### DISCUSSION TITLE
[bug] Nextjs Class extends value undefined Immediate on Import

### DISCUSSION THREAD

**Discussion by benjamin852** [plhery/node-twitter-api-v2#320] [Q&A]
**Describe the bug**

**To Reproduce**
I am trying to indicate this library with my nextjs app but it is not working right off of installation crashes on the first import statement.

My code is super basic just trying to query a tweet.

```
// const TwitterApi = require('twitter-api-v2');
import { TwitterApi } from 'twitter-api-v2';

// Instanciate with desired auth type (here's Bearer v2 auth)
const twitterClient = new TwitterApi(process.env.TWITTER_BEARER);

// Tell typescript it's a readonly app
const roClient = twitterClient.readOnly;
export const twitterApiGetTweetInfo = async (id, res) => {
  try {
    const tweetData = await roClient.v2.singleTweet(id, {
      expansions: ['public_metrics'],
    });
    console.log(tweetData, 'the tweeet');
    return res.status(200).json(tweetData);
  } catch (error) {
    console.log('Error On twitterApiGetTweetInfo():', error);
    return error.statusCode;
  }
};
```
**Expected behavior**

I am hoping to be able to query a tweet when the function is called

**Version**
 - Node.js version: v12.18.3
 - Lib version: "1.12.2"
 - OS (especially if you use Windows): MacOS Monterey

**Additional context**

Again I think this has something to do with me using nextjs (v12.0.3) even if i comment all my code out and leave the import statement the app still crashes. 
The error that I am getting is 
```
Uncaught TypeError: Class extends value undefined is not a constructor or null
    at module.exports (keccak.js?c404:3:1)
```

Worth noting before i was receiving a different error 
```
error - ./node_modules/@pedrouid/iso-crypto/dist/cjs/helpers/env/node.js:7:0
Module not found: Can't resolve 'crypto'
```

But i was able to resolve this by adding 
```
config.resolve = {
      ...config.resolve,
      fallback: {
        fs: false,
        path: false,
        os: false,
        crypto: false,
        http: false,
        https: false,
        zlib: false,
      },
    };
```
to my next.config file

i had this error for each of these separately the crypto error was just one example so i did false for each that it appeared and that solved this problem.

The constructor error message is now what is blocking me.

Thank you

**Comment by alkihis** [comment]
Did you tried to import it with:

```
const { TwitterApi } = require('twitter-api-v2');
```
(note the braces around variable declaration)

---

This error means there is a circular import somewhere.
As the lib correctly functions for other users, the problem is to look for circular imports in your code.

**Comment by benjamin852** [comment]
Hi @alkihis yes i did try, same error.

**Comment by alkihis** [comment]
Please install manually `events` package from NPM, and specify `fallback: { events: false }`.

**Comment by benjamin852** [comment]
hi added to my fallback still the same error. thank you for the suggestion

**Comment by benjamin852** [comment]
One thing worth noting. If I remove the fallback{XXX: false} variables i have set in my next.config file the Class extends value undefined error goes away. So that is what is causing the error i think

The issue is if I don't have this then I get 
```
Module not found: Can't resolve 'fs'
```

Which after googling makes sense because fs is not available in the browser.

But I am importing the twitter-api-v2 library inside my ./pages/api directory which I thought would work fine with nodejs functionality, but I am clearly wrong about this. If anyone knows where I can most import and query the twitter api from within a nextjs app then I think that would solve my error since I would then not need to change up the config file which will fix the extends value undefined error.

**Comment by alkihis** [comment]
As this isn't an issue with the lib and more a configuration issue of how use Node.js libraries in Next, I convert this issue as a discussion :)
